Pre-release workflow issue with v16-beta

## Current behavior

As I'm following along the defined [pre-release workflow steps](https://github.com/semantic-release/semantic-release/blob/v16.0.0-beta.31%40beta/docs/recipes/pre-releases.md) to test out my CI workflow I've run across an issue during the step where I [publish my beta release to the default distribution channel (`master`)](https://github.com/semantic-release/semantic-release/blob/v16.0.0-beta.31%40beta/docs/recipes/pre-releases.md#publishing-the-300-beta-release-to-the-default-distribution-channel). Once I merge `beta` to `master` and push changes up, a Github Action workflow is kicked off to run semantic-release. Unfortunately, it fails to finish with the following error:

```sh
EINVALIDNEXTVERSION The release `5.0.0` on branch `master` cannot be published as it is out of range.
```

I've posted the full semantic-release logs to better illustrate what is happening. Not sure if I did something incorrectly, the docs are incorrect, or there's an issue with `semantic-release@v16-beta`.

**NOTE:** This is on a private repo so I can't link you to that. If absolutely needed, I can create a public repo to try and simulate this issue. Here are some screenshots for more info:

**List of current branches in Github:**
<img width="288" alt="Screen Shot 2019-11-22 at 13 08 31" src="https://user-images.githubusercontent.com/757524/69457056-6f478600-0d29-11ea-882d-f8fb629b8e1e.png">

**List of current tags in Github:**
<img width="288" alt="Screen Shot 2019-11-22 at 13 08 42" src="https://user-images.githubusercontent.com/757524/69457080-7e2e3880-0d29-11ea-85e8-5830d76373b7.png">

**List of current releases in NPM:**
<img width="754" alt="Screen Shot 2019-11-22 at 13 09 13" src="https://user-images.githubusercontent.com/757524/69457114-93a36280-0d29-11ea-8a70-f7d93651329c.png">

## Expected behavior

Expected once `beta` was merged into `master` and pushed, semantic-release would successfully release a new major.

## Environment

**semantic-release** version: ^16.0.0-beta

CI environment: Github Actions using `ubuntu-latest`

Plugins used: See package.json config below 👇 

**semantic-release** configuration:

In package.json:
```json
"release": {
    "plugins": [
      "@semantic-release/commit-analyzer",
      "@semantic-release/release-notes-generator",
      "@semantic-release/changelog",
      "@semantic-release/npm",
      "@semantic-release/git"
    ]
  },
```

I'm using the following Github Actions for this:
```yaml
- name: Semantic Release
  id: semantic # Need an `id` for output variables
  uses: cycjimmy/semantic-release-action@v2
  with:
    branch: master
    semantic_version: ^16.0.0-beta
    extra_plugins: |
      @semantic-release/git@beta
      @semantic-release/changelog
  env:
    GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
    NODE_AUTH_TOKEN: ${{ secrets.NPM_TOKEN }}
    NPM_TOKEN: ${{ secrets.NPM_TOKEN }}
```

CI logs:
```sh
[7:42:03 PM] [semantic-release] â€º â„¹  Running semantic-release version 16.0.0-beta.35
[7:42:03 PM] [semantic-release] â€º âœ”  Loaded plugin "verifyConditions" from "@semantic-release/changelog"
[7:42:03 PM] [semantic-release] â€º âœ”  Loaded plugin "verifyConditions" from "@semantic-release/npm"
[7:42:03 PM] [semantic-release] â€º âœ”  Loaded plugin "verifyConditions" from "@semantic-release/git"
[7:42:03 PM] [semantic-release] â€º âœ”  Loaded plugin "analyzeCommits" from "@semantic-release/commit-analyzer"
[7:42:03 PM] [semantic-release] â€º âœ”  Loaded plugin "generateNotes" from "@semantic-release/release-notes-generator"
[7:42:03 PM] [semantic-release] â€º âœ”  Loaded plugin "prepare" from "@semantic-release/changelog"
[7:42:03 PM] [semantic-release] â€º âœ”  Loaded plugin "prepare" from "@semantic-release/npm"
[7:42:03 PM] [semantic-release] â€º âœ”  Loaded plugin "prepare" from "@semantic-release/git"
[7:42:03 PM] [semantic-release] â€º âœ”  Loaded plugin "publish" from "@semantic-release/npm"
[7:42:03 PM] [semantic-release] â€º âœ”  Loaded plugin "addChannel" from "@semantic-release/npm"
[7:42:04 PM] [semantic-release] â€º âœ”  Run automated release from branch master
[7:42:05 PM] [semantic-release] â€º âœ”  Allowed to push to the Git repository
[7:42:05 PM] [semantic-release] â€º â„¹  Start step "verifyConditions" of plugin "@semantic-release/changelog"
[7:42:05 PM] [semantic-release] â€º âœ”  Completed step "verifyConditions" of plugin "@semantic-release/changelog"
[7:42:05 PM] [semantic-release] â€º â„¹  Start step "verifyConditions" of plugin "@semantic-release/npm"
[7:42:05 PM] [semantic-release] [@semantic-release/npm] â€º â„¹  Verify authentication for registry https://registry.npmjs.org/
[7:42:05 PM] [semantic-release] [@semantic-release/npm] â€º â„¹  Reading npm config from /home/runner/work/_temp/.npmrc
my-private-repo
[7:42:05 PM] [semantic-release] â€º âœ”  Completed step "verifyConditions" of plugin "@semantic-release/npm"
[7:42:05 PM] [semantic-release] â€º â„¹  Start step "verifyConditions" of plugin "@semantic-release/git"
[7:42:05 PM] [semantic-release] â€º âœ”  Completed step "verifyConditions" of plugin "@semantic-release/git"
[7:42:05 PM] [semantic-release] â€º â„¹  Found git tag v4.1.1 associated with version 4.1.1 on branch master
[7:42:05 PM] [semantic-release] â€º â„¹  Found 5 commits since last release
[7:42:05 PM] [semantic-release] â€º â„¹  Start step "analyzeCommits" of plugin "@semantic-release/commit-analyzer"
[7:42:05 PM] [semantic-release] [@semantic-release/commit-analyzer] â€º â„¹  Analyzing commit: Merge branch 'beta'
[7:42:05 PM] [semantic-release] [@semantic-release/commit-analyzer] â€º â„¹  The commit should not trigger a release
[7:42:05 PM] [semantic-release] [@semantic-release/commit-analyzer] â€º â„¹  Analyzing commit: chore(release): 5.0.0-beta.2 [skip ci]

# [5.0.0-beta.2](https://github.com/my-private-repo/test-singlerepo/compare/v5.0.0-beta.1@beta...v5.0.0-beta.2@beta) (2019-11-22)

### Features

* minor feature on [@beta](https://github.com/beta) ([5e890a2](https://github.com/my-private-repo/test-singlerepo/commit/5e890a27f1501c66d3eae54372eee27853d3b5b5))
[7:42:05 PM] [semantic-release] [@semantic-release/commit-analyzer] â€º â„¹  The commit should not trigger a release
[7:42:05 PM] [semantic-release] [@semantic-release/commit-analyzer] â€º â„¹  Analyzing commit: feat: minor feature on @beta
[7:42:05 PM] [semantic-release] [@semantic-release/commit-analyzer] â€º â„¹  The release type for the commit is minor
[7:42:05 PM] [semantic-release] [@semantic-release/commit-analyzer] â€º â„¹  Analyzing commit: chore(release): 5.0.0-beta.1 [skip ci]

# [5.0.0-beta.1](https://github.com/my-private-repo/test-singlerepo/compare/v4.1.0...v5.0.0-beta.1@beta) (2019-11-22)

### Features

* first feature on beta ([e0081b6](https://github.com/my-private-repo/test-singlerepo/commit/e0081b6144dd26ae1bd94cfb10cf88364e5125f0))

### BREAKING CHANGES

* it breaks something
[7:42:05 PM] [semantic-release] [@semantic-release/commit-analyzer] â€º â„¹  The commit should not trigger a release
[7:42:05 PM] [semantic-release] [@semantic-release/commit-analyzer] â€º â„¹  Analyzing commit: feat: first feature on beta

BREAKING CHANGE: it breaks something
[7:42:05 PM] [semantic-release] [@semantic-release/commit-analyzer] â€º â„¹  The release type for the commit is major
[7:42:05 PM] [semantic-release] [@semantic-release/commit-analyzer] â€º â„¹  Analysis of 5 commits complete: major release
[7:42:05 PM] [semantic-release] â€º âœ”  Completed step "analyzeCommits" of plugin "@semantic-release/commit-analyzer"
[7:42:06 PM] [semantic-release] â€º â„¹  The next release version is 5.0.0
[7:42:06 PM] [semantic-release] â€º âœ–  EINVALIDNEXTVERSION The release `5.0.0` on branch `master` cannot be published as it is out of range.
Based on the releases published on other branches, only versions within the range >=4.1.1 <5.0.0 can be published from branch master.

The following commits are responsible for the invalid release:

    * Merge branch 'beta' (1edc7a1)
    * chore(release): 5.0.0-beta.2 [skip ci] (32aeea1)
    * feat: minor feature on @beta (5e890a2)
    * chore(release): 5.0.0-beta.1 [skip ci] (6185875)
    * feat: first feature on beta (e0081b6)

Those commits should be moved to a valid branch with git merge (https://git-scm.com/docs/git-merge) or git cherry-pick (https://git-scm.com/docs/git-cherry-pick) and removed from branch master with git revert (https://git-scm.com/docs/git-revert) or git reset (https://git-scm.com/docs/git-reset).

A valid branch could be next.

See the workflow configuration documentation (https://github.com/semantic-release/semantic-release/blob/master/docs/usage/workflow-configuration.md) for more details.

##[error]SemanticReleaseError: The release `5.0.0` on branch `master` cannot be published as it is out of range.
##[error]Node run failed with exit code 1
```

**Most recent commits from master:**
```sh
1edc7a11c67813bf158e641c331e63f046c1228f (HEAD -> master, origin/master, origin/HEAD) Merge branch 'beta'
7f9a4c4591850236843d6d5f77a672580c1b35f5 (tag: v4.1.1) chore(release): 4.1.1 [skip ci]
32aeea1201b15d5e9de6e24a2592793eb27c64b9 (tag: v5.0.0-beta.2@beta, origin/beta) chore(release): 5.0.0-beta.2 [skip ci]
31c01f4e045b43eeb292d6c721212992393f0537 fix: bug fix for v4.x.x
5e890a27f1501c66d3eae54372eee27853d3b5b5 feat: minor feature on @beta
6185875e16c33f8032a6d4049235fd70812e4b73 (tag: v5.0.0-beta.1@beta) chore(release): 5.0.0-beta.1 [skip ci]
e0081b6144dd26ae1bd94cfb10cf88364e5125f0 feat: first feature on beta
c44e232c34fe71e71101a835483da60a6ce41a8c (tag: v4.1.0@next, tag: v4.1.0, origin/next, next) chore(release): 4.1.0 [skip ci]
d780573d31810f86b37ebd3d3874a6e5fa6e5133 Merge branch 'master' into next
461bb393da2cfb097382391b60622aa031ea1786 (tag: v3.2.0) chore(release): 3.2.0 [skip ci]
44169c40c66930adb853fabf021bdc67c73f8876 feat: a small feat in latest 3.x.x
e6c4b10941efcb1631b670fe3b6f87297708b664 (tag: v4.0.1@next, tag: v4.0.1) chore(release): 4.0.1 [skip ci]
```

--- a/docs/usage/workflow-configuration.md
+++ b/docs/usage/workflow-configuration.md
@@ -100,7 +100,7 @@ For example the configuration `['master', {name: 'pre/rc', prerelease: '${name.r
 
 ### Release branches
 
-A release branch is the base type of branch used by **semantic-release** that allows to publish releases with a [semantic version](https://semver.org), optionally on a specific distribution channel. Distribution channels (for example [npm dist-tags](https://docs.npmjs.com/cli/dist-tag) or [Chrome release channels](https://www.chromium.org/getting-involved/dev-channel)) are a way to distribute new releases only to a subset of users in order to get early feedback. Later on those releases can be added to the general distribution channel to be made available to all users.
+A release branch is the base type of branch used by **semantic-release** that allows to publish releases with a [semantic version](https://semver.org), optionally on a specific distribution channel. Distribution channels (for example [npm dist-tags](https://docs.npmjs.com/cli/dist-tag) or [Chrome release channels](https://www.chromium.org/getting-involved/dev-channel)) are a way to distribute new releases only to a subset of users in order to get early feedback. Later on, those releases can be added to the general distribution channel to be made available to all users.
 
 **semantic-release** will automatically add releases to the corresponding distribution channel when code is [merged from a release branch to another](#merging-into-a-release-branch).
 
@@ -112,20 +112,17 @@ See [publishing on distribution channels recipe](../recipes/distribution-channel
 
 #### Pushing to a release branch
 
-With the configuration `"branches": ["master", "next"]`, if the last release published from `master` is `1.0.0` then:
+With the configuration `"branches": ["master", "next"]`, if the last release published from `master` is `1.0.0` and the last one from `next` is `2.0.0` then:
 - Only versions in range `1.x.x` can be published from `master`, so only `fix` and `feat` commits can be pushed to `master`
-- Only versions in range `>=2.0.0` release can be published from `next`, so a `BREAKING CHANGE` commit must be pushed first on `next` and can be followed by any type of commits
+- Once `next` get merged into `master` the release `2.0.0` will be made available on the channel associated with `master` and both `master` and `next` will accept any commit type
 
-With the configuration `"branches": ["master", "next", "next-major"]`, if the last release published from `master` is `1.0.0` then:
-- Only versions in range `1.0.x` can be published from `master`, so only `fix` commits can be pushed to `master`
-- Only versions in range `>=1.1.0` can be published from `next`, so a `feat` commit must be pushed first on `next` and can be followed by `fix` and `feat` commits
-- Only versions in range `>=2.0.0` release can be published from `next`, so a `BREAKING CHANGE` commit must be pushed first on `next-major` and can be followed by any type of commits
-
-Those verifications prevent situations such as:
-1. Create a `feat` commit on `next` which triggers the release of version `1.0.0` on channel `next`
+This verification prevent scenario such as:
+1. Create a `feat` commit on `next` which triggers the release of version `1.0.0` on the `next` channel
 2. Merge `next` into `master` which adds `1.0.0` on the default channel
-3. Push a `fix` commit to `master` which triggers the release of version `1.0.1` on the default channel
-4. Push a `fix` commit to `next` which would attempt to release the version `1.0.1` on channel `next` and fails as this version already exists
+3. Create a `feat` commit on `next` which triggers the release of version `1.1.0` on the `next` channel
+4. Create a `feat` commit on `master` which would attempt to release the version `1.1.0` on the default channel
+
+In step 4 **semantic-release** will throw an `EINVALIDNEXTVERSION` error to prevent the attempt at releasing version `1.1.0` which was already released on step 3 with a different codebase. The error will indicate that the commit should be created on `next` instead. Alternatively if the `next` branch is merged into `master`, the version `1.1.0` will be made available on the default channel and the `feat` commit would be allowed on `master` to release `1.2.0`.
 
 #### Merging into a release branch
 

--- a/lib/branches/normalize.js
+++ b/lib/branches/normalize.js
@@ -1,5 +1,4 @@
 const {sortBy, isNil} = require('lodash');
-const semver = require('semver');
 const semverDiff = require('semver-diff');
 const {FIRST_RELEASE, RELEASE_TYPE} = require('../definitions/constants');
 const {
@@ -61,35 +60,29 @@ function release({release}) {
     return release;
   }
 
-  const breakpoints = release.length > 2 ? ['minor', 'major'] : ['major'];
-
-  // The intial bound is the last release from the base branch of `FIRST_RELEASE` (1.0.0)
-  let bound = getLatestVersion(tagsToVersions(release[0].tags)) || FIRST_RELEASE;
+  // The intial lastVersion is the last release from the base branch of `FIRST_RELEASE` (1.0.0)
+  let lastVersion = getLatestVersion(tagsToVersions(release[0].tags)) || FIRST_RELEASE;
 
   return release.map(({name, tags, channel, ...rest}, idx) => {
     const versions = tagsToVersions(tags);
-    // The lower bound is the highest version between the current branch last release and the previous branch upper bound (`bound`)
-    const min = highest(getLatestVersion(versions), bound);
-    if (release.length - 1 === idx) {
-      // If the current branch is the last one of the release branch, there is no upper bound
-      bound = undefined;
-    } else {
-      // The default upper bound is the lower bound increment with the release type of the current branch position
-      const upperBound = semver.inc(min, breakpoints[idx]);
-      // Find the lowest version that is present on the current branch but none of the previous ones
-      const nextFirstVersion = getFirstVersion(tagsToVersions(release[idx + 1].tags), release.slice(0, idx + 1));
-      // The upper bound is the lowest version between `nextFirstVersion` and the default upper bound
-      bound = lowest(nextFirstVersion, upperBound);
-    }
+    // The new lastVersion is the highest version between the current branch last release and the previous branch lastVersion
+    lastVersion = highest(getLatestVersion(versions), lastVersion);
+    // The upper bound is:
+    //  - None if the current branch is the last one of the release branches
+    //  - Otherwise, The upper bound is the lowest version that is present on the current branch but none of the previous ones
+    const bound =
+      release.length - 1 === idx
+        ? undefined
+        : getFirstVersion(tagsToVersions(release[idx + 1].tags), release.slice(0, idx + 1));
 
-    const diff = bound ? semverDiff(min, bound) : null;
+    const diff = bound ? semverDiff(lastVersion, bound) : null;
     return {
       ...rest,
       channel: idx === 0 ? channel : isNil(channel) ? name : channel,
       tags,
       type: 'release',
       name,
-      range: getRange(min, bound),
+      range: getRange(lastVersion, bound),
       accept: bound ? RELEASE_TYPE.slice(0, RELEASE_TYPE.indexOf(diff)) : RELEASE_TYPE,
       main: idx === 0,
     };

--- a/test/branches/branches.test.js
+++ b/test/branches/branches.test.js
@@ -30,9 +30,9 @@ test('Enforce ranges with branching release workflow', async t => {
   }));
   t.is(getBranch(result, '1.0.x').range, '>=1.0.0 <1.0.0', 'Cannot release on 1.0.x before a releasing on master');
   t.is(getBranch(result, '1.x').range, '>=1.1.0 <1.0.0', 'Cannot release on 1.x before a releasing on master');
-  t.is(getBranch(result, 'master').range, '>=1.0.0 <1.1.0', 'Can release only patch on master');
-  t.is(getBranch(result, 'next').range, '>=1.1.0 <2.0.0', 'Can release only minor on next');
-  t.is(getBranch(result, 'next-major').range, '>=2.0.0', 'Can release only major on next-major');
+  t.is(getBranch(result, 'master').range, '>=1.0.0');
+  t.is(getBranch(result, 'next').range, '>=1.0.0');
+  t.is(getBranch(result, 'next-major').range, '>=1.0.0');
 
   release(branches, 'master', '1.0.0');
   result = (await getBranches('repositoryUrl', 'master', {options: {branches}})).map(({name, range}) => ({
@@ -41,14 +41,18 @@ test('Enforce ranges with branching release workflow', async t => {
   }));
   t.is(getBranch(result, '1.0.x').range, '>=1.0.0 <1.0.0', 'Cannot release on 1.0.x before a releasing on master');
   t.is(getBranch(result, '1.x').range, '>=1.1.0 <1.0.0', 'Cannot release on 1.x before a releasing on master');
-  t.is(getBranch(result, 'master').range, '>=1.0.0 <1.1.0', 'Can release only patch on master');
+  t.is(getBranch(result, 'master').range, '>=1.0.0');
+  t.is(getBranch(result, 'next').range, '>=1.0.0');
+  t.is(getBranch(result, 'next-major').range, '>=1.0.0');
 
   release(branches, 'master', '1.0.1');
   result = (await getBranches('repositoryUrl', 'master', {options: {branches}})).map(({name, range}) => ({
     name,
     range,
   }));
-  t.is(getBranch(result, 'master').range, '>=1.0.1 <1.1.0', 'Can release only patch, > than 1.0.1 on master');
+  t.is(getBranch(result, 'master').range, '>=1.0.1', 'Can release only > than 1.0.1 on master');
+  t.is(getBranch(result, 'next').range, '>=1.0.1', 'Can release only > than 1.0.1 on next');
+  t.is(getBranch(result, 'next-major').range, '>=1.0.1', 'Can release only > than 1.0.1 on next-major');
 
   merge(branches, 'master', 'next');
   merge(branches, 'master', 'next-major');
@@ -56,9 +60,9 @@ test('Enforce ranges with branching release workflow', async t => {
     name,
     range,
   }));
-  t.is(getBranch(result, 'master').range, '>=1.0.1 <1.1.0', 'Can release only patch, > than 1.0.1 on master');
-  t.is(getBranch(result, 'next').range, '>=1.1.0 <2.0.0', 'Can release only minor on next');
-  t.is(getBranch(result, 'next-major').range, '>=2.0.0', 'Can release only major on next-major');
+  t.is(getBranch(result, 'master').range, '>=1.0.1', 'Can release only > than 1.0.1 on master');
+  t.is(getBranch(result, 'next').range, '>=1.0.1', 'Can release only > than 1.0.1 on next');
+  t.is(getBranch(result, 'next-major').range, '>=1.0.1', 'Can release only > than 1.0.1 on next-major');
 
   release(branches, 'next', '1.1.0');
   release(branches, 'next', '1.1.1');
@@ -66,7 +70,9 @@ test('Enforce ranges with branching release workflow', async t => {
     name,
     range,
   }));
-  t.is(getBranch(result, 'next').range, '>=1.1.1 <2.0.0', 'Can release only patch or minor, > than 1.1.0 on next');
+  t.is(getBranch(result, 'master').range, '>=1.0.1 <1.1.0', 'Can release only patch, > than 1.0.1 on master');
+  t.is(getBranch(result, 'next').range, '>=1.1.1', 'Can release only > than 1.1.1 on next');
+  t.is(getBranch(result, 'next-major').range, '>=1.1.1', 'Can release > than 1.1.1 on next-major');
 
   release(branches, 'next-major', '2.0.0');
   release(branches, 'next-major', '2.0.1');
@@ -74,6 +80,8 @@ test('Enforce ranges with branching release workflow', async t => {
     name,
     range,
   }));
+  t.is(getBranch(result, 'master').range, '>=1.0.1 <1.1.0', 'Can release only patch, > than 1.0.1 on master');
+  t.is(getBranch(result, 'next').range, '>=1.1.1 <2.0.0', 'Can release only patch or minor, > than 1.1.0 on next');
   t.is(getBranch(result, 'next-major').range, '>=2.0.1', 'Can release any version, > than 2.0.1 on next-major');
 
   merge(branches, 'next-major', 'beta');
@@ -88,7 +96,6 @@ test('Enforce ranges with branching release workflow', async t => {
 
   merge(branches, 'master', '1.0.x');
   merge(branches, 'master', '1.x');
-  release(branches, 'master', '1.0.1');
   result = (await getBranches('repositoryUrl', 'master', {options: {branches}})).map(({name, range}) => ({
     name,
     range,
@@ -121,8 +128,9 @@ test('Enforce ranges with branching release workflow', async t => {
     name,
     range,
   }));
-  t.is(getBranch(result, 'master').range, '>=1.1.1 <1.2.0', 'Can release only patch, > than 1.1.1 on master');
-  t.is(getBranch(result, 'next').range, '>=1.2.0 <2.0.0', 'Can release only patch or minor, > than 1.2.0 on next');
+
+  t.is(getBranch(result, 'master').range, '>=1.1.1', 'Can release only > than 1.1.1 on master');
+  t.is(getBranch(result, 'next').range, '>=1.1.1 <2.0.0', 'Can release only patch or minor, > than 1.1.1 on next');
   t.is(getBranch(result, 'next-major').range, '>=2.0.1', 'Can release any version, > than 2.0.1 on next-major');
   t.is(
     getBranch(result, '1.0.x').range,
@@ -136,7 +144,7 @@ test('Enforce ranges with branching release workflow', async t => {
     name,
     range,
   }));
-  t.is(getBranch(result, 'master').range, '>=1.1.1 <1.2.0', 'Can release only patch, > than 1.1.1 on master');
+  t.is(getBranch(result, 'master').range, '>=1.1.1', 'Can release only > than 1.1.1 on master');
   t.is(getBranch(result, '1.0.x').range, '>=1.0.4 <1.1.0', 'Can release on 1.0.x only within range');
   t.is(getBranch(result, '1.x').range, '>=1.1.0 <1.1.0', 'Cannot release on 1.x before >= 2.0.0 is released on master');
 
@@ -145,7 +153,7 @@ test('Enforce ranges with branching release workflow', async t => {
     name,
     range,
   }));
-  t.is(getBranch(result, 'master').range, '>=1.1.1 <1.2.0', 'Can release only patch, > than 1.1.1 on master');
+  t.is(getBranch(result, 'master').range, '>=1.1.1', 'Can release only > than 1.1.1 on master');
   t.is(getBranch(result, '1.0.x').range, '>=1.0.4 <1.1.0', 'Can release on 1.0.x only within range');
   t.is(getBranch(result, '1.x').range, '>=1.1.1 <1.1.1', 'Cannot release on 1.x before >= 2.0.0 is released on master');
 
@@ -155,19 +163,20 @@ test('Enforce ranges with branching release workflow', async t => {
     name,
     range,
   }));
-  t.is(getBranch(result, 'master').range, '>=2.0.1 <2.1.0', 'Can release only patch, > than 2.0.1 on master');
-  t.is(getBranch(result, 'next').range, '>=2.1.0 <3.0.0', 'Can release only minor on next');
-  t.is(getBranch(result, 'next-major').range, '>=3.0.0', 'Can release only major on next-major');
+  t.is(getBranch(result, 'master').range, '>=2.0.1', 'Can release only > than 2.0.1 on master');
+  t.is(getBranch(result, 'next').range, '>=2.0.1', 'Can release only > than 2.0.1 on next');
+  t.is(getBranch(result, 'next-major').range, '>=2.0.1', 'Can release only > than 2.0.1 on next-major');
   t.is(getBranch(result, '1.x').range, '>=1.1.1 <2.0.0', 'Can release on 1.x only within range');
 
   merge(branches, 'beta', 'master');
+  release(branches, 'master', '3.0.0');
   result = (await getBranches('repositoryUrl', 'master', {options: {branches}})).map(({name, range}) => ({
     name,
     range,
   }));
-  t.is(getBranch(result, 'master').range, '>=2.0.1 <2.1.0', 'Can release only patch, > than 2.0.1 on master');
-  t.is(getBranch(result, 'next').range, '>=2.1.0 <3.0.0', 'Can release only minor on next');
-  t.is(getBranch(result, 'next-major').range, '>=3.0.0', 'Can release only major on next-major');
+  t.is(getBranch(result, 'master').range, '>=3.0.0', 'Can release only > than 3.0.0 on master');
+  t.is(getBranch(result, 'next').range, '>=3.0.0', 'Can release only > than 3.0.0 on next');
+  t.is(getBranch(result, 'next-major').range, '>=3.0.0', 'Can release only > than 3.0.0 on next-major');
 
   branches.push({name: '1.1.x', tags: []});
   merge(branches, '1.x', '1.1.x');

--- a/test/branches/normalize.test.js
+++ b/test/branches/normalize.test.js
@@ -159,19 +159,26 @@ test('Release branches - initial state', t => {
       .release({release})
       .map(({type, name, range, accept, channel, main}) => ({type, name, range, accept, channel, main})),
     [
-      {type: 'release', name: 'master', range: '>=1.0.0 <1.1.0', accept: ['patch'], channel: undefined, main: true},
+      {
+        type: 'release',
+        name: 'master',
+        range: '>=1.0.0',
+        accept: ['patch', 'minor', 'major'],
+        channel: undefined,
+        main: true,
+      },
       {
         type: 'release',
         name: 'next',
-        range: '>=1.1.0 <2.0.0',
-        accept: ['patch', 'minor'],
+        range: '>=1.0.0',
+        accept: ['patch', 'minor', 'major'],
         channel: 'next',
         main: false,
       },
       {
         type: 'release',
         name: 'next-major',
-        range: '>=2.0.0',
+        range: '>=1.0.0',
         accept: ['patch', 'minor', 'major'],
         channel: 'next-major',
         main: false,
@@ -293,15 +300,15 @@ test('Release branches - Handle missing previous tags in branch history', t => {
       {
         type: 'release',
         name: 'master',
-        range: '>=2.0.0 <3.0.0',
-        accept: ['patch', 'minor'],
+        range: '>=2.0.0',
+        accept: ['patch', 'minor', 'major'],
         channel: undefined,
         main: true,
       },
       {
         type: 'release',
         name: 'next',
-        range: '>=3.0.0',
+        range: '>=2.0.0',
         accept: ['patch', 'minor', 'major'],
         channel: 'next',
         main: false,
@@ -310,40 +317,7 @@ test('Release branches - Handle missing previous tags in branch history', t => {
   );
 });
 
-test('Release branches - enforce release gaps after downstream merge', t => {
-  const release = [
-    {name: 'master', tags: toTags(['1.0.0', '1.1.0', '2.0.0'])},
-    {name: 'next', tags: toTags(['1.0.0', '1.1.0', '2.0.0'])},
-    {name: 'next-major', tags: toTags(['1.0.0', '1.1.0', '2.0.0'])},
-  ];
-
-  t.deepEqual(
-    normalize
-      .release({release})
-      .map(({type, name, range, accept, channel, main}) => ({type, name, range, accept, channel, main})),
-    [
-      {type: 'release', name: 'master', range: '>=2.0.0 <2.1.0', accept: ['patch'], channel: undefined, main: true},
-      {
-        type: 'release',
-        name: 'next',
-        range: '>=2.1.0 <3.0.0',
-        accept: ['patch', 'minor'],
-        channel: 'next',
-        main: false,
-      },
-      {
-        type: 'release',
-        name: 'next-major',
-        range: '>=3.0.0',
-        accept: ['patch', 'minor', 'major'],
-        channel: 'next-major',
-        main: false,
-      },
-    ]
-  );
-});
-
-test('Release branches - limit releases on 2nd and 3rd branche based on 1st branch last release', t => {
+test('Release branches - limit releases on 2nd and 3rd branch based on 1st branch last release', t => {
   const release = [
     {name: 'master', tags: toTags(['1.0.0', '1.1.0', '2.0.0', '3.0.0'])},
     {name: 'next', tags: toTags(['1.0.0', '1.1.0'])},
@@ -355,19 +329,26 @@ test('Release branches - limit releases on 2nd and 3rd branche based on 1st bran
       .release({release})
       .map(({type, name, range, accept, channel, main}) => ({type, name, range, accept, channel, main})),
     [
-      {type: 'release', name: 'master', range: '>=3.0.0 <3.1.0', accept: ['patch'], channel: undefined, main: true},
+      {
+        type: 'release',
+        name: 'master',
+        range: '>=3.0.0',
+        accept: ['patch', 'minor', 'major'],
+        channel: undefined,
+        main: true,
+      },
       {
         type: 'release',
         name: 'next',
-        range: '>=3.1.0 <4.0.0',
-        accept: ['patch', 'minor'],
+        range: '>=3.0.0',
+        accept: ['patch', 'minor', 'major'],
         channel: 'next',
         main: false,
       },
       {
         type: 'release',
         name: 'next-major',
-        range: '>=4.0.0',
+        range: '>=3.0.0',
         accept: ['patch', 'minor', 'major'],
         channel: 'next-major',
         main: false,

--- a/test/index.test.js
+++ b/test/index.test.js
@@ -97,8 +97,8 @@ test('Plugins are called with expected values', async t => {
     {
       channel: undefined,
       name: 'master',
-      range: '>=1.0.0 <2.0.0',
-      accept: ['patch', 'minor'],
+      range: '>=1.0.0',
+      accept: ['patch', 'minor', 'major'],
       tags: [{channels: ['next'], gitTag: 'v1.0.0', version: '1.0.0'}],
       type: 'release',
       main: true,
@@ -106,7 +106,7 @@ test('Plugins are called with expected values', async t => {
     {
       channel: 'next',
       name: 'next',
-      range: '>=2.0.0',
+      range: '>=1.0.0',
       accept: ['patch', 'minor', 'major'],
       tags: [{channels: ['next'], gitTag: 'v1.0.0', version: '1.0.0'}],
       type: 'release',

